src/packages/@support/features/feedback-section/layout/FeedbackSection.svelte

```diff
@@ -9,51 +9,70 @@
   import { SearchIcon } from "$lib/assets/app.asset";
   import { Select } from "@library/forms";
   import { CategoryIcon, StatusIcon } from "@library/icons";
-  import { FeedbackType } from "@support/common/types/feedback";
+  import {
+    FeedbackType,
+    FeedbackStatusType,
+  } from "@support/common/types/feedback";
   export let type = FeedbackType.ALL_CATEGORY;
   import { tickIcon } from "@library/forms/select/svgs";
   export let onInputFeedback;
   export let onAddFeedback;
   export let fetchPosts;
   export let onRetrievePost;
-  let currentSort = "newest";
-  let posts = [];
-  let filteredPosts = [];
+
   let searchTerm = "";
+
+  let currentSort = "trending";
+  let posts = [];
   let userInfo: any = {};
   user.subscribe((value) => {
     userInfo = value;
   });
   let id = "";
-  const getPosts = async (sortType) => {
-    currentSort = sortType; // Added: Update the sorting type
-    const listPosts = await fetchPosts(sortType); // Modified: Pass sortType to fetchPosts
+  const getPosts = async (
+    sortType: string,
+    searchQuery: string = "",
+    status: string,
+  ) => {
+    currentSort = sortType;
+    const listPosts = await fetchPosts(sortType, searchQuery, status);
     posts = listPosts?.data?.posts;
   };
 
-  // const filterPosts = () => {
-  //   filteredPosts = posts.filter((post) =>
-  //     post.title.toLowerCase().includes(searchTerm.toLowerCase()),
-  //   );
-  // };
-
-  // const handleSearch = (event) => {
-  //   searchTerm = event.target.value;
-  //   filterPosts();
-  // };
+  let status = "open,under review,planned,in progress,complete";
 
   onMount(async () => {
-    getPosts(currentSort);
+    getPosts(currentSort, searchTerm, status);
   });
   let isPostopen = false;
 
-  const filterPosts = () => {
-    if (searchTerm) {
-      filteredPosts = posts.filter((post) =>
-        post.title.toLowerCase().includes(searchTerm.toLowerCase()),
-      );
+  // Function to handle input change
+  const handleInputChange = async (searchQuery: string) => {
+    searchTerm = searchQuery;
+    await getPosts(currentSort, searchQuery, status); // Fetch posts with search term
+  };
+
+  const handleSelectChange = async (_status: string) => {
+    if (_status == "all status") {
+      status = "open,under review,planned,in progress,complete";
```

Comment:
store this in a const string, and use at multiple places

Reply:
Resolved